tests/lib/common.py

```diff
@@ -49,18 +49,23 @@ def decode_wrapper(i):
 
 
 def wait_for_result(func, *args, matcher=simple_matcher(True), attempts=20,
-                    interval=5, decode=decode_wrapper, **kwargs):
+                    interval=5, decode=decode_wrapper, catch_exceptions=True,
+                    **kwargs):
     """Runs `func` with `args` until `matcher(out)` returns true or timesout
 
     Returns the matching result, or raises an exception.
     """
 
     for i in range(attempts):
-        out = func(*args, **kwargs)
-        if decode:
-            out = decode(out)
-        if matcher(out):
-            return out
+        try:
+            out = func(*args, **kwargs)
+            if decode:
+                out = decode(out)
+            if matcher(out):
+                return out
+        except Exception as e:
+            if catch_exceptions and i == (attempts-1):
```

Comment:
This has taken `catch_exceptions` to mean "log them". Instead, if `catch_exceptions` is False, I'd like to raise the Exception. eg:

```
except Exception as e:
  if not catch_exceptions:
    raise
  if i == (attempts - 1):
    logger.error(e.output)
```

Reply:
thinking about it, would it be clearer if we called it `ignore_exceptions` or something else instead?

Reply:
I would go for ignore_exceptions defaulting to True, and still logging the last one in all cases

Reply:
```
          except Exception as e:                                                     
              if ignore_exceptions:                                                  
                  if i == (attempts-1):                                              
                      logger.error(e.output)                                         
              else:                                                                  
                  logger.error(e.output)                                             
                  break
```

Reply:
Instead of weird calculation `i == (attempts-1):` you might use reversed range for this loop. Also, we can report user how many tries left during while waiting for the result.

Reply:
I'm fine with that, but we should still `raise` rather than break.

Reply:
@brunoleon are you going to address the comments?

Reply:
Just pushed some changes fixing above comments